pkg/hostpath/controllerserver.go

```diff
@@ -46,7 +45,10 @@ type controllerServer struct {
 	caps []*csi.ControllerServiceCapability
 }
 
-func NewControllerServer() *controllerServer {
+func NewControllerServer(ephemeral bool) *controllerServer {
+	if ephemeral {
+		return &controllerServer{caps: getControllerServiceCapabilities(nil)}
```

Comment:
Does Kubernetes require a controller service? I thought the CSI spec allowed them to be optional?

Reply:
@kfox1111 it turns out, some calls from the Controller service are fired by either the registrar or k8s, cant remember, to determine Controller Capability.